appender.go

```diff
@@ -38,12 +38,19 @@ import (
 	"golang.org/x/sync/errgroup"
 )
 
+type ProcessedStatus string
```

Comment:
```suggestion
type processedStatus string
```

I don't think this needs to be exported - let's avoid exporting things unless needed.

Reply:
I think we can just delete the type though - see other comments.